lightyear/src/utils/ready_buffer.rs

```diff
@@ -15,28 +15,34 @@ pub struct ReadyBuffer<K, T> {
     pub heap: BinaryHeap<ItemWithReadyKey<K, T>>,
 }
 
-impl<K: Ord, T: PartialEq> Default for ReadyBuffer<K, T> {
+impl<K: Ord, T> Default for ReadyBuffer<K, T> {
     fn default() -> Self {
         Self {
             heap: BinaryHeap::default(),
         }
     }
 }
 
-impl<K: Ord, T: PartialEq> ReadyBuffer<K, T> {
+impl<K: Ord, T> ReadyBuffer<K, T> {
     pub fn new() -> Self {
         Self {
             heap: BinaryHeap::default(),
         }
     }
 }
 
-impl<K: Ord + Clone, T: PartialEq> ReadyBuffer<K, T> {
+impl<K: Ord + Clone, T> ReadyBuffer<K, T> {
     /// Adds an item to the heap marked by time
     pub fn push(&mut self, key: K, item: T) {
         self.heap.push(ItemWithReadyKey { key, item });
     }
 
+    /// Returns a reference to the item with the highest `K` value or None if the queue is empty.
+    /// Does not remove the item from the queue.
+    pub fn peek_latest_item(&self) -> Option<(K, &T)> {
```

Comment:
Wouldn't this return the item with the lowest K value?

Reply:
According to the docs, `std::collections::BinaryHeap` is a max-heap: https://doc.rust-lang.org/std/collections/struct.BinaryHeap.html#method.peek

I've added a unit test for the new function. Based on the other functions' comments it looks some of them assume the buffer is a min-heap.

Reply:
Yes but I inverted the `Ord` order of `ItemWithReadyKey` to make it a min-heap, or I thought..
I'll have to take a closer look